ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/KsqlArray.java

```diff
@@ -282,7 +282,7 @@ public KsqlArray add(final Boolean value) {
    * @return a reference to this
    */
   public KsqlArray add(final BigDecimal value) {
-    delegate.add(value);
+    delegate.add(value.doubleValue());
```

Comment:
The Vert.x JSON types don't accept `BigDecimal`. For example,
```
new JsonArray().add(new BigDecimal(11.1));
```
results in
```
java.lang.IllegalStateException: Illegal type in JsonObject: class java.math.BigDecimal

	at io.vertx.core.json.JsonObject.checkAndCopy(JsonObject.java:1072)
	at io.vertx.core.json.JsonArray.add(JsonArray.java:446)
```

Reply:
What if the BigDecimal is added as a String? Will the row get properly deserialized on the server before inserting?

Reply:
Pretty sure the answer is no but I'll answer more definitely in a follow-up PR -- I've got one in the works to add server and client integration tests for inserting types such as decimal, array, map, etc (and fix a bug the new tests exposed) :)

Going to merge this and https://github.com/confluentinc/ksql/pull/5456 first to avoid merge hell with myself.

Reply:
Hey @purplefox I stand corrected! This PR adds integration test coverage for inserting decimals (and also arrays and maps) and works fine if we serialize BigDecimal as a string instead of a double: https://github.com/confluentinc/ksql/pull/5469

As mentioned in the PR description, the drawback is that we now have an inconsistency where the decimal type is treated differently in terms of allowable casts, compared to the other numeric types. For example:
```
final KsqlObject obj = new KsqlObject().put("f1", new BigDecimal("1.13")).put("f2", 12.23);
System.out.println(obj.getString("f1")); // This succeeds
System.out.println(obj.getString("f2")); // This fails with "java.lang.ClassCastException: java.lang.Double cannot be cast to java.lang.CharSequence"
```

I think the drawback is acceptable so I've made the change.